cpp/mg_utility/mg_utils.hpp

```diff
@@ -101,34 +105,55 @@ class OnScopeExit {
+///@brief Maps the Memgraph in-memory graph to a user-defined graph view.
+/// Supports building unweighted and weighted graph views, as well as views of
+/// subgraphs with given nodes and relationships.
+/// Node/relationship IDs are zero-indexed.
+/// The graph view holds connection information and local<->Memgraph ID mapping.
 ///
-///@param graph Memgraph graph
-///@param result Memgraph result object
-///@param memory Memgraph storage object
-///@param graph_type Graph directedness
-///@return mg_graph::Graph
+///@param graph -- Memgraph graph
+///@param result -- Memgraph result object
+///@param memory -- Memgraph storage object
+///@param graph_type -- graph directedness
+///@param weighted -- graph weightedness
+///@param weight_property -- relationship property containing weight
+///@param default_weight -- default relationship weight if property not set
+///@param subgraph -- if set, limit the graph view to subgraph
+///@param subgraph_nodes -- nodes selected for subgraph
+///@param subgraph_relationships -- relationships selected for subgraph
 ///
+///@return mg_graph::Graph -- graph view
 template <typename TSize = std::uint64_t>
-std::unique_ptr<mg_graph::Graph<TSize>> GetGraphView(mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory,
-                                                     const mg_graph::GraphType graph_type) {
+std::unique_ptr<mg_graph::Graph<TSize>> GraphViewGetter(
+    mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory, const mg_graph::GraphType graph_type,
+    const bool weighted = false, const char *weight_property = "weight", double default_weight = 1.0,
+    const bool subgraph = false, mgp_list *subgraph_nodes = nullptr, mgp_list *subgraph_relationships = nullptr) {
   auto graph = std::make_unique<mg_graph::Graph<TSize>>();
 
+  std::unordered_set<std::uint64_t> subgraph_node_ids, subgraph_edge_ids;
+  if (subgraph) {
+    for (std::size_t i = 0; i < mgp::list_size(subgraph_nodes); i++) {
+      subgraph_node_ids.insert(mgp::vertex_get_id(mgp::value_get_vertex(mgp::list_at(subgraph_nodes, i))).as_int);
+    }
+    for (std::size_t i = 0; i < mgp::list_size(subgraph_relationships); i++) {
+      subgraph_edge_ids.insert(mgp::edge_get_id(mgp::value_get_edge(mgp::list_at(subgraph_relationships, i))).as_int);
+    }
+  }
+
   ///
   /// Mapping Memgraph in-memory vertices into graph view
   ///
 
   // Safe vertices iterator creation
-
   auto *vertices_it = mgp::graph_iter_vertices(memgraph_graph, memory);
   mg_utility::OnScopeExit delete_vertices_it([&vertices_it] { mgp::vertices_iterator_destroy(vertices_it); });
 
   // Iterate through Memgraph vertices and map them to GraphView
   for (auto *vertex = mgp::vertices_iterator_get(vertices_it); vertex;
        vertex = mgp::vertices_iterator_next(vertices_it)) {
+    if (subgraph && subgraph_node_ids.find(mgp::vertex_get_id(vertex).as_int) == subgraph_node_ids.end()) {
+      continue;
+    }
```

Comment:
Maybe we don't need this here, this seems like a costly operation and if we only save a reference to vertex instead of `vertix_id`, everything would be fine, right?

Reply:
As `vertex_get_id` is already called in `CreateGraphNode`, would it make sense to move the vertex existence check inside that function?

Reply:
I think you should only traverse the nodes provided by the function input. However, this check of inclusiveness might still be relevant when traversing edges too since both source and destination nodes must be in the subgraph.